crates/bevy_render/src/render_resource/storage_buffer.rs

```diff
@@ -78,6 +101,25 @@ impl<T: ShaderType + WriteInto> StorageBuffer<T> {
     }
 }
 
+/// Stores data to be transferred to the GPU and made accessible to shaders as a dynamic storage buffer.
+///
+/// Dynamic storage buffers can be made available to shaders as some combination of read/write, unlike
+/// [`UniformBuffer`](crate::render_resource::UniformBuffer) which is read-only. Furthermore, dynamic storage buffers
+/// can store much larger data than uniform buffers, which are best suited to relatively small data. Dynamic storage buffers
+/// are particularly well-suited to storing like a Rust-vector (wgpu's "runtime-sized arrays") and have a
+/// [`push`](crate::render_resource::DynamicStorageBuffer::push) method, unlike
+/// [`StorageBuffer`](crate::render_resource::StorageBuffer). Note however that
+/// WebGL2 does not support dynamic storage buffers, so other alternatives to consider are vertex/instance buffers (see
+/// [`BufferVec`](crate::render_resource::BufferVec)), or data textures ([`Texture`](crate::render_resource::Texture)),
+/// depending on what is most appropriate for the use case.
```

Comment:
```suggestion
/// Dynamic storage buffers can be made available to shaders as some combination of read/write, unlike
/// [`UniformBuffer`](crate::render_resource::UniformBuffer) which is read-only. Furthermore, dynamic storage buffers
/// can store much larger data than uniform buffers, which are only guaranteed to be up to 16kB per binding. Dynamic storage buffers
/// support multiple separate bindings at dynamic byte offsets and so have a
/// [`push`](crate::render_resource::DynamicStorageBuffer::push) method, unlike
/// [`StorageBuffer`](crate::render_resource::StorageBuffer). Note however that
/// WebGL2 does not support dynamic storage buffers, so other alternatives to consider are vertex/instance buffers (see
/// [`BufferVec`](crate::render_resource::BufferVec)), or data textures ([`Texture`](crate::render_resource::Texture)),
/// depending on what is most appropriate for the use case.
```

Reply:
This feels like a fairly weak motivation for dynamic storage buffers but I'm not really sure when I would want to use them. Perhaps for multiple separate batches of things where there is some batch data and then a runtime-sized array of instance data.